x-pack/plugins/reporting/public/management/report_listing.tsx

```diff
@@ -90,49 +92,86 @@ class ReportListingUi extends Component<Props, State> {
       capabilities,
       config,
     } = this.props;
-    const ilmLocator = urlService.locators.get('ILM_LOCATOR_ID');
-    const hasIlmPolicy = ilmPolicyContextValue.status !== 'policy-not-found';
-    const showIlmPolicyLink = Boolean(ilmLocator && hasIlmPolicy);
-    return (
-      <>
-        <EuiPageHeader
-          data-test-subj="reportingPageHeader"
-          bottomBorder
-          pageTitle={
-            <FormattedMessage id="xpack.reporting.listing.reportstitle" defaultMessage="Reports" />
-          }
-          description={
-            <FormattedMessage
-              id="xpack.reporting.listing.reports.subtitle"
-              defaultMessage="Get reports generated in Kibana applications."
-            />
-          }
-        />
+    if (config.statefulSettings.enabled) {
```

Comment:
There is some slightly deep conditional complexity in ReportListingUi that can be hidden. Instead of checking for `statefulSettings.enabled` in this component, put this code in the `ReportListingStateful` component.

To do that, ReportListingStateful and ReportListingDefault should render the page header and footer themselves:

```
ReportListing: already doing the right thing
  - if statefulSettings.enabled
    - render ReportListingStateful
  - else
    - render ReportListingDefault

ReportListingStateful: should also render the page sections and ILM controls
  - call useIlmPolicyStatus()
  - render the page header
  - render ILM controls
  - render ReportListingUi
  - render footer

ReportListingDefault: should also render the page sections
  - render the page header
  - render ReportListingUi
  - render footer

ReportListingUi: should just render the table
  - no longer renders page header and footer
  - the renderTable() method should become the render() method
```